ui/src/data_explorer/components/SendToDashboardOverlay.tsx

```diff
@@ -0,0 +1,156 @@
+import React, {PureComponent} from 'react'
+
+import {getNewDashboardCell} from 'src/dashboards/utils/cellGetters'
+
+import {
+  OverlayContainer,
+  OverlayHeading,
+  OverlayBody,
+  MultiSelectDropdown,
+} from 'src/reusable_ui'
+
+import {addDashboardCellAsync} from 'src/dashboards/actions'
+
+import {QueryConfig, Dashboard, Source} from 'src/types'
+
+interface Props {
+  queryConfig: QueryConfig
+  dashboards: Dashboard[]
+  source: Source
+  rawText: string
+  onCancel: () => void
+  addDashboardCell: typeof addDashboardCellAsync
+}
+
+interface State {
+  selectedIDs: string[]
+  hasQuery: boolean
+  name: string
+}
+
+class SendToDashboardOverlay extends PureComponent<Props, State> {
+  constructor(props) {
+    super(props)
+    const {queryConfig} = this.props
+
+    this.state = {
+      selectedIDs: [],
+      hasQuery: queryConfig.fields.length !== 0,
+      name: '',
+    }
+  }
+
+  public handleChangeName = e => {
+    const name = e.target.value
+    this.setState({name})
+  }
+
+  public render() {
+    const {onCancel} = this.props
+    const {hasQuery, name} = this.state
+
+    return (
+      <OverlayContainer>
+        <OverlayHeading title="Send to Dashboard" />
+        <OverlayBody>
+          <div className="form-group">
+            <label htmlFor="New Cell Name"> New Cell Name </label>
```

Comment:
had never seen `for` on a `label` before, that's cool. is it for screen readers?

Reply:
I think it is, yeah